### DISCUSSION TITLE
Generating Gabor patches during the experiment

### DISCUSSION THREAD

**Discussion by prajwalgatti** [jspsych/jsPsych#756] [Q&A] [answered]
Hi all,

I am new to jsPsych, and I am trying to create a task that includes two Gabor patches being presented to the user during a trial. 

I am wondering if there is a way to generate them on-the-go, or before a trial, like in PsychtoolBox.

Any suggestions/advice?

**Comment by jodeleeuw** [comment, marked as answer]
There's no built-in method for Gabor patches, but you could certainly incorporate a separate library for generating the patches into your experiment. This should allow you to generate them on demand.

This library looks like a promising one to hook into: https://github.com/jtth/gaborgen-js

**Reply by prajwalgatti** [reply]
Thanks, Josh

I looked into that library, and the [readme](https://github.com/jtth/gaborgen-js#gaborgen) mentions that it takes about 500ms to generate a Gabor patch. Do you think this would be a problem as I am presenting the stimulus for about 200ms mid-trial?  Would this cause any delays in the experiment?

**Reply by jodeleeuw** [reply]
It looks like the library generates a base 64 encoded image. You could generate these ahead of time (during loading) and then reference them during the experiment. Depending on the flow of your experiment you might also be able to generate them during some other wait time.

**Reply by prajwalgatti** [reply]
I see. I guess that could work, I will try that, thank you.

**Reply by prajwalgatti** [reply]
Hey @jodeleeuw, 

I want to generate some basic cues and fixation dot on the go. My reason behind this is -- for varying screen sizes, I would account for the resolution and produce them before the start of the experiment. 

There is also a simple pie-chart I would need to present at the end of the trial as feedback based on the trial performance. 

Would you suggest using something like Snap SVG? or canvas element of HTML? 

What would be a good way to go about this without causing any delays in expt?
